python_modules/dagster/dagster/_core/storage/state/base.py

```diff
@@ -0,0 +1,113 @@
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import TYPE_CHECKING, ClassVar, Optional
+
+from dagster._core.instance import MayHaveInstanceWeakref, T_DagsterInstance
+from dagster._core.storage.state.state_versions import StateVersionInfo, StateVersions
+from dagster._time import get_current_timestamp
+
+if TYPE_CHECKING:
+    from dagster._core.storage.runs.base import RunStorage
+    from dagster._core.storage.state.run_storage_state_storage import RunStorageStateStorage
+
+
+class StateStorage(ABC, MayHaveInstanceWeakref[T_DagsterInstance]):
```

Comment:
I think some qualifier makes sense since in the context of the instance runs etc are also "state"

DefsStateStorage
ComponentStateStorage
DefinitionStateStorage

Reply:
went with DefsStateStorage